### ISSUE TITLE
motion / doorbell / two way audio homekit

### ISSUE THREAD

**Issue by bagobones** [AlexxIT/go2rtc#669]
Trying to keep latency as low as possible (sometimes worse than 15s in HA) having go2rtc export cameras to everything seems to be much faster and efficient than lets say going camera--->go2rtc-->homeassistant--> homekit 

Any chance there are plans to support motion / doorbell type "EVENTS" passed on with the exposed homekit cameras or a functional 2 way feed with homekit to use the native GUI on apple devices? Or is that outside the project scope and better for something like HomeAssistant to handle (although the video processing sucks)?

I could see mapping like for like ONVIF-->homekit, homekit-->homekit being easier to support than say having MQTT / Webhooks being mapped.  Or maybe a API that frigate could use for its existing person / motion / new custom events.

Being able to connect / map sensors for motion / doorbell would be very cool however.

**Comment by AlexxIT** [comment]
What model of camera are you talking about?

**Comment by bagobones** [comment]
Well I officially have the Reolink WiFi doorbell now... I have not been very happy with the mapping or video stability of passing exposing cameras from home assistant TO homekit

Would be very cool if go2rtc could map ONVIF doorbells back out as homekit doorbells, might require a config syntax that maps the sensors / events for motion and button (visitor) if they are not standard.

**Comment by digaus** [comment]
> Well I officially have the Reolink WiFi doorbell now... I have not been very happy with the mapping or video stability of passing exposing cameras from home assistant TO homekit
> 
> Would be very cool if go2rtc could map ONVIF doorbells back out as homekit doorbells, might require a config syntax that maps the sensors / events for motion and button (visitor) if they are not standard.

Why not use Scrypted?

**Comment by bagobones** [comment]
> Why not use Scrypted?

Seems redundant.. I am already running frigate which contains go2rtc. There is a lot of similar functionality in the tools but frigate/go2rtc already do most of what I want other than being able to link sensors.

I tried Scrypted a while ago before finding go2rtc and I prefer the clean config structure of go2rtc.. I found some of Scrypteds modularity actually confusing to use.

Given some of the great work on getting a number of 2way audio cameras working, going further and exposing these to homekit with 2-way audio, motion and doorbell support.

**Comment by pfeifix** [comment]
Is there any update on this?
I would also be very happy if go2rtc supports a way to augment a HomeKit camery with a doorbell trigger switch. This has some useful advantages (as described above).
Unfortunately I cannot be done with Scrypted as it does not have support for importing HomeKit streams or provide a HomeKit proxy like go2rtc does.

**Comment by pfeifix** [comment]
> Please let me know if the problem is relevant

Yes, it is. It would be a superb thing if the doorbell profile could be added to my native HomeKit camera which is the Logitech Circle View (outdoor) but unfortunately it has no built-in doorbell functionality.

Can the feature be added to go2rtc?

**Comment by pfeifix** [comment]
> This is very big task. Not in near future

OK, too bad. But motion and luminance sensor are already connected? I would assume it is just an additional profile to add for the doorbell button to appear in HomeKit?

**Comment by AlexxIT** [comment]
It's even bigger task :)
I think now you using your camera in HomeKit transparent proxy mode. Go2rtc don't make changes to it spec.
This issue about generic camera that should be converted to HomeKit camera with additional features.
The problem is - go2rtc don't have any events logic in its core yet.

**Comment by pfeifix** [comment]
I see. So there seems currently no solution for the doorbell to be added to a HomeKit camera. Scrypted is not able to import HomeKit cameras at all and via the RTSP/ffmpeg import (together with the stream provided by go2rtc) the connection is not stable, too (and there is no 2-way audio). 

If you find the time to implement the feature please report back. This would be kind of a game changer for me.

**Comment by JoshuaSeidel** [comment]
> It's even bigger task :)
> 
> I think now you using your camera in HomeKit transparent proxy mode. Go2rtc don't make changes to it spec.
> 
> This issue about generic camera that should be converted to HomeKit camera with additional features.
> 
> The problem is - go2rtc don't have any events logic in its core yet.

@AlexxIT Could you guide me to do this in proxy mode? I'm utilizing the HomeKit feature of go2rtc in frigate

**Comment by AlexxIT** [comment]
https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#module-homekit

**Comment by noelhibbard** [comment]
First, thank you for go2rtc — the HomeKit proxying (native HomeKit cameras) and exposing non-HomeKit cameras to HomeKit is very well done.

One missing piece, which several users have mentioned, is **motion event triggering** in HomeKit.

### Scope clarification
I am **not** asking go2rtc to perform motion detection itself.  
Motion detection is already handled by **Frigate** in my setup.

What would be very useful is a **simple external trigger** that can raise a HomeKit motion event.

### Existing working example
The `homebridge-camera-ffmpeg` plugin exposes a small HTTP endpoint.  
When it receives an HTTP `POST`, it raises a HomeKit motion event, which then triggers notifications on iOS and tvOS.

In my setup:
- A small app listens to Frigate MQTT events (e.g., person detected on porch)
- When motion is detected, it sends an HTTP `POST` to `homebridge-camera-ffmpeg`
- HomeKit then generates motion notifications reliably

### Proposed approach for go2rtc
Instead of implementing motion detection, go2rtc could expose a simple HTTP API, for example:

- `POST /api/motion/start`
- `POST /api/motion/stop`

When called, this would trigger the HomeKit motion characteristic for the associated camera.

You could reference `homebridge-camera-ffmpeg` to see how the HAP motion characteristic is handled.

### Benefits
- Keeps go2rtc focused on streaming and HomeKit integration
- Avoids duplicating motion detection logic
- Allows Frigate (or other NVRs / scripts) to integrate cleanly
- Enables native iOS and tvOS motion notifications

This would make go2rtc a very flexible HomeKit bridge for advanced camera setups.


### Yes, I used ChatGPT to write this because the layers of technologies are very confusing and I am terrible at articulating stuff like this in words. What I originally wrote was way too confusing but with literally zero context, ChatGPT understood my intent and flawlessly rewrote it.

**Comment by skywalker1979** [comment]
Got caught into this as well. I'm now using go2rtc for my analog doorbell....mixing 2 way audio coming from the building gate with the video flow from a camera.  (i compiled the PR 2259 to get 2 way audio) and it works flawlessly.
The only thing missing now is being able to associate the "camera/audio mix" to the events of the doorbell.
 What a great piece of software this is!!

**Comment by hiSweid** [comment]
Tested PR #2259 personally against a real Hikvision video doorbell (outdoor station, DS-KV/DS-KH family) and can confirm it works ? two-way audio, talking and listening at the same time, no issues.
